### PR TITLE
Don't redirect to the 404 page on error

### PR DESCRIPTION
**Pull request by eatonphil**
This also fixes some bugs in the setError and fetch code. Closes https://github.com/linode/manager/issues/1234

### CODE REVIEW

File: src/actions/errors.js [hunk 1]
```diff
@@ -8,15 +8,17 @@ export const TOGGLE_DETAILS = '@@errors/TOGGLE_DETAILS';
...
-    const type = response.headers && response.headers.get('Content-Type');
-    const _json = !json && type === 'application/json'
-      ? await response.json() : null;
-    dispatch({
-      type: SET_ERROR,
-      status: response.statusCode,
-      statusText: response.statusText,
-      json: _json,
-    });
+    if (response.status) {
```
Comment (mparke) [inline src/actions/errors.js]: when does this not have `response.status`?

Comment (eatonphil) [inline src/actions/errors.js]: setError is called on any exception, not just request failures :/

Comment (mparke) [inline src/actions/errors.js]: ah...if we need to revisit make a follow up?

Comment (eatonphil) [inline src/actions/errors.js]: hmm... yeah. It doesn't exactly need to change and dealing with more errors ties more into issues that do exist.

Comment (mparke) [inline src/actions/errors.js]: well, this is sort of problematic, because if it's not a response object with status this just swallows the error call?

Comment (mparke) [inline src/actions/errors.js]: 🙏  https://github.com/linode/manager/issues/1585

Comment (eatonphil) [inline src/actions/errors.js]: mhm sounds good

**Pull request merged**